src/Compilers/CSharp/Portable/Symbols/SynthesizedNamespaceSymbol.cs

```diff
@@ -0,0 +1,78 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Immutable;
+using Roslyn.Utilities;
+using System.Diagnostics;
+using System;
+using Microsoft.CodeAnalysis.Emit;
+
+namespace Microsoft.CodeAnalysis.CSharp.Symbols
+{
+    /// <summary>
+    /// Synthesized namespace that contains synthesized types or subnamespaces.
+    /// All its members are stored in a table on <see cref="CommonPEModuleBuilder"/>.
+    /// </summary>
+    internal sealed class SynthesizedNamespaceSymbol : NamespaceSymbol, IEquatable<SynthesizedNamespaceSymbol>
```

Comment:
>IEquatable<SynthesizedNamespaceSymbol> [](start = 72, length = 38)

Who uses this?

Reply:
I followed the pattern of MissingNamespaceSymbol, which implements equality.

---
In reply to: [302225353](https://github.com/dotnet/roslyn/pull/36854#discussion_r302225353) [](ancestors = 302225353)

Reply:
If there are no consumers of this interface, we should remove it.

---
In reply to: [302230781](https://github.com/dotnet/roslyn/pull/36854#discussion_r302230781) [](ancestors = 302230781,302225353)

Reply:
Should I remove Equals and GetHashCode impl as well? If not then we should keep the interface, since that's a standard pattern how to implement equality.

---
In reply to: [302274715](https://github.com/dotnet/roslyn/pull/36854#discussion_r302274715) [](ancestors = 302274715,302230781,302225353)

Reply:
We should remove the interface implementation only. No `Symbol` types implement `IEquatable<>`.

---
In reply to: [302289323](https://github.com/dotnet/roslyn/pull/36854#discussion_r302289323) [](ancestors = 302289323,302274715,302230781,302225353)

Reply:
OK. Sounds like a bug though.

---
In reply to: [302290074](https://github.com/dotnet/roslyn/pull/36854#discussion_r302290074) [](ancestors = 302290074,302289323,302274715,302230781,302225353)